### ISSUE TITLE
Improve standard JSON output for "contract methods"

### ISSUE THREAD

**Issue by axic** [ethereum/solidity#3596]
Currently there are two outputs listing contract methods:
```
          // The list of function hashes
          methodIdentifiers: {
            "delegate(address)": "5c19a95c"
          },
          // Function gas estimates
          gasEstimates: {
            creation: {
              codeDepositCost: "420000",
              executionCost: "infinite",
              totalCost: "infinite"
            },
            external: {
              "delegate(address)": "25000"
            },
            internal: {
              "heavyLifting()": "infinite"
            }
          }
```

It would be nice to have a single object with more information, which could also handle unimplemented functions (see #3594).

Option 1:
```
methods: {
  // parentheses since that is invalid as a function name
  "creation": {
  },
  "fallback": {
  },
  "external": {
    "delegate(address)": {
      selector: "5c19a95c",
      gas: "25000"
    }
  },
  unimplemented: [
    "unimplementedFunction"
  ],
  internal: {
    "heavyLifting()": {
      gas: "infinite"
    }
  }
}
```

Option 2:
```
methods: {
  // parentheses since that is invalid as a function name
  // creation or constructor
  "(creation)": {
  },
  "(fallback)": {
  },
  "delegate(address)": {
    selector: "5c19a95c",
    visibility: "external",
    gas: "25000"
  },
  "unimplementedFunction":  {
    selector: "44228182",
    type: "unimplemented"
  },
  "heavyLifting()": {
    visibility: "internal",
    gas: "infinite"
  }
}
```

**Comment by axic** [comment]
After #3198 there will be a need to distinguish between "receive ether" and "wildcard".

**Comment by axic** [comment]
@chriseth I think we should do this improvement step, could be done fairly quickly after we agree on the new format.

**Comment by chriseth** [comment]
The problem is that we actually cannot really identify internal functions by their name. Should we rather use the AST id?
